src/@apollo/client/core/ApolloClient__Core_Types.res

```diff
@@ -170,19 +170,17 @@ module MutationQueryReducersMap = {
     // }> = {
     //     [queryName: string]: MutationQueryReducer<T>;
     // };
-    type t<'jsData> = Js.Dict.t<MutationQueryReducer.Js_.t<'jsData>>
+    type t<'jsData> = RescriptCore.Dict.t<MutationQueryReducer.Js_.t<'jsData>>
   }
 
-  type t<'data> = Js.Dict.t<MutationQueryReducer.t<'data>>
+  type t<'data> = RescriptCore.Dict.t<MutationQueryReducer.t<'data>>
 
   let toJs: (t<'data>, ~safeParse: Types.safeParse<'data, 'jsData>) => Js_.t<'jsData> = (
     t,
     ~safeParse,
   ) =>
-    Js.Dict.map(
-      (. mutationQueryReducer) => mutationQueryReducer->MutationQueryReducer.toJs(~safeParse),
```

Comment:
I'm pretty sure this `Js.Dict.map` issue you described in the PR description is from the return type of this function rather than an issue with `Js.Dict.map` itself, for context this works in #160 :

```rescript
    Js.Dict.map(
      (. mutationQueryReducer) =>
        (. json, options) =>
          MutationQueryReducer.toJs(. mutationQueryReducer, ~safeParse, json, options),
      t,
    )
```

I've had some issues with RescriptCore as a dependency in libraries; because the compiler de-duplicates the package versions, you can't really be sure of the version of RescriptCore being used by the consuming application. We could add a peer dependency to `^1.3.0` to at least make sure it's an 11-only + uncurried mode compatible version--though I'm a bit curious about how the versioning would actually work with it being included in the compiler later on

Reply:
> I was under the impression that Js was deprecated in Rescript ≥11.

I'm not sure if it's been formally deprecated yet but yeah it's definitely on the way out. Personally I'd choose to avoid adding `@rescript/core` as a dependency in a library until it's in a released version of the compiler and support for the previous versions is dropped. That said, hopefully `RescriptCore.Dict.map` is fleshed out at this point and won't be broken

Reply:
I agree this is probably the way to go instead of adding another dependency. As soon as core is included in the compiler we can do a pass and swap all the `Belt` calls as well.